### PR TITLE
feat(tm2): Add additional unit tests for tm2/pkg/std

### PR DESCRIPTION
**Pull request by linhpn99**
Add necessary unit tests to increase test coverage for this package

<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

**Comment by leohhhn** [comment]
https://github.com/gnolang/gno/pull/2333#issuecomment-2343808931

File: tm2/pkg/std/gasprice_test.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+)
+
+func TestParseGasPrice(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name     string
+		input    string
+		expected GasPrice
+		hasError bool
```
Comment (thehowl) [inline tm2/pkg/std/gasprice_test.go:16]: better errorIs / errorContains

File: tm2/pkg/std/gasprice_test.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+func TestParseGasPrices(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name     string
+		input    string
+		expected []GasPrice
+		hasError bool
```
Comment (thehowl) [inline tm2/pkg/std/gasprice_test.go:74]: same here

File: tm2/pkg/std/tx_test.go [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+			if tc.expectedError == "" {
+				require.NoError(t, err)
+			} else {
+				require.Error(t, err, tc.expectedError)
+			}
+		})
+	}
+}
+
+func Test_CountSubKeys(t *testing.T) {
```
Comment (thehowl) [inline tm2/pkg/std/tx_test.go:123]: Why underscore?

File: tm2/pkg/std/gasprice_test.go
Comment (thehowl) [inline tm2/pkg/std/gasprice_test.go:1]: This is deadcode, but ok

File: tm2/pkg/std/tx_test.go [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+	fee := NewFee(1000, Coin{Denom: "atom", Amount: 10})
+	sigs := []Signature{}
+	tx := NewTx(msgs, fee, sigs, "test memo")
+	chainID := "test-chain"
+	accountNumber := uint64(1)
+	sequence := uint64(1)
+
+	signBytes, err := tx.GetSignBytes(chainID, accountNumber, sequence)
+	require.NoError(t, err)
+	require.NotEmpty(t, signBytes)
```
Comment (thehowl) [inline tm2/pkg/std/tx_test.go:215]: should be a golden test (match signBytes against a known example output)

Comment (linhpn99) [reply tm2/pkg/std/tx_test.go]: https://github.com/gnolang/gno/pull/2739/commits/5e22fe43bd1ae19e4bbdea61b24f42bffa19e6ee

File: tm2/pkg/std/tx_test.go (hunk 3 above)
Comment (linhpn99) [reply tm2/pkg/std/tx_test.go]: https://github.com/gnolang/gno/pull/2739/commits/5e22fe43bd1ae19e4bbdea61b24f42bffa19e6ee

File: tm2/pkg/std/gasprice_test.go (hunk 2 above)
Comment (linhpn99) [reply tm2/pkg/std/gasprice_test.go]: https://github.com/gnolang/gno/pull/2739/commits/5e22fe43bd1ae19e4bbdea61b24f42bffa19e6ee

File: tm2/pkg/std/gasprice_test.go (hunk 1 above)
Comment (linhpn99) [reply tm2/pkg/std/gasprice_test.go]: https://github.com/gnolang/gno/pull/2739/commits/5e22fe43bd1ae19e4bbdea61b24f42bffa19e6ee

**Pull request closed without merging**